core/src/main/java/com/aem/demo/core/models/MainPageModel.java

```diff
@@ -50,83 +48,50 @@ public class MainPageModel {
-        }
-
-        String searchText = Optional.ofNullable(servletRequest).map(SlingHttpServletRequest::getRequestParameterMap)
-                .map(map -> map.getValue("search"))
-                .map(RequestParameter::getString)
-                .orElse(null);
-
-
-        String paginationNumber = Optional.ofNullable(servletRequest).map(SlingHttpServletRequest::getRequestParameterMap)
-                .map(map -> map.getValue("page"))
-                .map(RequestParameter::getString)
-                .orElse(null);
-
-        isNumeric(paginationNumber);
-
-
-        if (searchText != null) {
-            models = searchService.retrieveModels(searchText);
-            paginateModels(models);
-            if (paginationNumber != null) {
-                models = separateModels(models, Integer.parseInt(paginationNumber));
-            }
-        } else if (paginationNumber != null) {
-            paginateModels(holdModels);
-            models = separateModels(holdModels, Integer.parseInt(paginationNumber));
-        } else {
-            paginateModels(holdModels);
-            models = holdModels.subList(0, 5);
+            modelsList.add(jcrContent.adaptTo(NewsCardModel.class));
         }
+        return modelsList;
     }
 
     public String getPath() {
         return resource.getPath();
     }
 
-    private List<NewsCardModel> separateModels(List<NewsCardModel> list, int paginationNumber) {
-        int limit = Math.min(paginationNumber * 5, list.size());
-        List<NewsCardModel> separatedPart = new ArrayList<>();
-        for (int i = (paginationNumber - 1) * 5; i < limit; i++) {
-            separatedPart.add(list.get(i));
-        }
-        return separatedPart;
-    }
-
-    private void paginateModels(List<NewsCardModel> models) {
-        double listSize = models.size();
-        double listPart = listSize / 5;
-        double roundPart = Math.ceil(listPart);
-
-        pageNumbers = IntStream.range(1, (int) (roundPart + 1)).boxed().collect(Collectors.toList());
-
-    }
-
-    private void isNumeric(String string) {
-        int intValue;
-
-        if (string == null || string.equals("")) {
-            log.info("String cannot be parsed, it is null or empty.");
-        }
-
+    private List<NewsCardModel> separateModels(List<NewsCardModel> list, String paginationNumber) {
+        int pNumber = 1;
         try {
-            intValue = Integer.parseInt(string);
+            pNumber = Integer.parseInt(paginationNumber);
         } catch (NumberFormatException e) {
-            log.error("Input String cannot be parsed to Integer.");
+            log.error("Exception occurred while parsing request parameter " + e.getMessage());
         }
+        pNumber = (pNumber - 1) * CARDS_PERPAGE;
+        int limit = Math.min(pNumber + CARDS_PERPAGE, list.size());
+        return list.subList(pNumber, limit);
     }
 
+    private void paginateNumbers(List<NewsCardModel> models) {
+        if (models.size() > 4) {
```

Comment:
The same logic works for both `>4` and `<=4` cases, no difference. There's no need in `if`
Please, in cases that are not obvious to you, create a "Scratch" file with a simple one-class Java application, and check your logic.
![image](https://user-images.githubusercontent.com/48795460/170672708-e1a32195-0e81-4407-90b9-ab04c767e7ef.png)
![image](https://user-images.githubusercontent.com/48795460/170672890-8362fc1b-db82-405a-8de2-d9cbc5058263.png)